libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5297,6 +5328,13 @@ bool GCS_MAVLINK::try_send_message(const enum ap_message id)
 #endif // HAL_HIGH_LATENCY2_ENABLED
         break;
 
+    case MSG_HYGROMETER_STATUS:
+#if AP_HYGROMETER_ENABLED
+        CHECK_PAYLOAD_SIZE(HYGROMETER_SENSOR);
```

Comment:
This payload size check has to be done each time you go to send a message.  As it stands you're only checking to see if you've got room for one `HYGROMETER_SENSOR` message, and you might try to send two, leading to corruption.